manim/mobject/mobject.py

```diff
@@ -2675,19 +2675,46 @@ def throw_error_if_no_points(self):
                 )
 
     # About z-index
-    def set_z_index(self, z_index_value: Union[int, float]):
+    def set_z_index(
+        self,
+        z_index_value: Union[int, float],
```

Comment:
Just `float` is fine here.
```suggestion
        z_index_value: float,
```